test/controllers/lookup_keys_controller_test.rb

```diff
@@ -1,59 +1,79 @@
+  let(:new_value_params) { { 'match' => 'hostgroup=db', 'value' => '4', 'omit' => '0' } }
+  let(:destroy_params) { { 'id' => key_with_value.override_values.first.id, '_destroy' => '1' } }
+
+  test '#update adds valid override' do
+    assert_equal key.override_values.count, 0
+    params = { :override => true, :lookup_values_attributes => { '112' => new_value_params }}
+    patch :update, params: { :id => key.to_param, :dummy_lookup_key => params }, session: set_session_user
+    assert_redirected_to dummy_lookup_keys_path
+
+    assert_equal key.reload.override_values.count, 1
+    value = key.override_values.first
     assert_equal 'hostgroup=db', value.match
     assert_equal 4, value.value
   end
 
-  test 'patch delete' do
-    params = @base.merge(:lookup_values_attributes => @delete)
-    patch :update, params: { :id => "#{@key.id}-#{@key.key}", :puppetclass_lookup_key => params }, session: set_session_user
-    assert_redirected_to puppetclass_lookup_keys_path
-    assert_equal 0, @key.reload.override_values.count
+  test '#update deletes values' do
+    params = { :override => true, :lookup_values_attributes => { '0' => destroy_params }}
+    assert_difference -> { key_with_value.override_values.count }, -1 do
+      patch :update, params: { :id => key_with_value.to_param, :dummy_lookup_key => params }, session: set_session_user
+      assert_redirected_to dummy_lookup_keys_path
+    end
   end
 
-  test 'patch add and delete' do
-    params = @base.merge(:lookup_values_attributes => @create.merge(@delete))
-    patch :update, params: { :id => "#{@key.id}-#{@key.key}", :puppetclass_lookup_key => params }, session: set_session_user
-    assert_redirected_to puppetclass_lookup_keys_path
-    assert_equal @key.reload.override_values.count, 1
-    updated = @key.override_values.first
-    assert_equal 4, updated.value
-    assert_equal @value.match, updated.match
+  test '#update creates and deletes values simultanously' do
+    params = { :override => true, :lookup_values_attributes => { '0' => destroy_params, '112' => new_value_params }}
+    patch :update, params: { :id => key_with_value.to_param, :dummy_lookup_key => params }, session: set_session_user
+    assert_redirected_to dummy_lookup_keys_path
+    assert_equal key_with_value.reload.override_values.count, 1
+    new_val = key_with_value.override_values.first
+    assert_equal 4, new_val.value
+    assert_equal 'hostgroup=db', new_val.match
   end
 
-  test 'patch conflicting' do
-    create = {'1462788609699' => @create.values.first }
-    params = @base.merge(:lookup_values_attributes => @create.merge(@delete.merge(create)))
-    patch :update, params: { :id => "#{@key.id}-#{@key.key}", :puppetclass_lookup_key => params }, session: set_session_user
+  test '#update handle match conflict' do
+    # skip('Need to figure out unique validation for has_many :lookup_values')
```

Comment:
```suggestion
    # TODO: figure out unique validation for has_many :lookup_values'
```

Reply:
It's fixed, I forgot to remove the comment ^_^

Reply:
I thought this was relevant to tests :)